src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/ApplicationPropertyPage/ApplicationManifestKindValueProvider.cs

```diff
@@ -0,0 +1,128 @@
+        ///     - It's the value "CustomManifest" which means the user will supply the path to a custom manifest file.
+        ///     - It's the value "NoManifest" which means the application doesn't have a manifest.
+        ///     - It's the value "DefaultManifest" which means that the application will have a default manifest.
+        ///
+        /// These three values map to two MSBuild properties - ApplicationManifest (for the first case) or NoWin32Manifest
+        /// which is true for the second case and false or non-existent for the third.
+        /// </remarks>
+        public override async Task<string> OnGetEvaluatedPropertyValueAsync(string propertyName, string evaluatedPropertyValue, IProjectProperties defaultProperties)
+        {
+            if (!string.IsNullOrEmpty(await defaultProperties.GetEvaluatedPropertyValueAsync(ApplicationManifestMSBuildProperty)))
+            {
+                return CustomManifestValue;
+            }
+
+            string noManifestPropertyValue = await defaultProperties.GetEvaluatedPropertyValueAsync(NoManifestMSBuildProperty);
+            if (noManifestPropertyValue?.Equals("true", StringComparison.InvariantCultureIgnoreCase) == true)
+            {
+                return NoManifestValue;
+            }
+
+            // It doesn't matter if it is set to false or the value is not present. We default to "DefaultManifest" scenario.
+            return DefaultManifestValue;
+        }
+
+        /// <summary>
+        /// Sets the application manifest kind property
+        /// </summary>
+        public override async Task<string?> OnSetPropertyValueAsync(
+            string propertyName,
+            string? unevaluatedPropertyValue,
+            IProjectProperties defaultProperties,
+            IReadOnlyDictionary<string, string>? dimensionalConditions = null)
+        {
+            if (string.Equals(unevaluatedPropertyValue, DefaultManifestValue, StringComparison.InvariantCultureIgnoreCase))
+            {
+                await SaveCurrentApplicationManifestValueAsync(defaultProperties);
+
+                await defaultProperties.DeletePropertyAsync(ApplicationManifestMSBuildProperty);
+                await defaultProperties.DeletePropertyAsync(NoManifestMSBuildProperty);
+            }
+            else if (string.Equals(unevaluatedPropertyValue, NoManifestValue, StringComparison.InvariantCultureIgnoreCase))
+            {
+                await SaveCurrentApplicationManifestValueAsync(defaultProperties);
+
+                await defaultProperties.DeletePropertyAsync(ApplicationManifestMSBuildProperty);
+                await defaultProperties.SetPropertyValueAsync(NoManifestMSBuildProperty, "true");
+            }
+            else if (string.Equals(unevaluatedPropertyValue, CustomManifestValue, StringComparison.InvariantCultureIgnoreCase))
+            {
+                await RestoreApplicationManifestValueAsync(defaultProperties);
```

Comment:
@drewnoakes What do you think of this approach of storing the previous values of properties when we clear them, and then restoring them later when/if the user re-enables them?

Reply:
I haven't reviewed the code yet but I like the idea in principle. I worry that users won't expect toggling settings to have destructive changes.

Reply:
Yeah, that was my motivation: you toggle a check box or change a drop down just to see what the other options are, and now you've lost your setting.

Reply:
Took a deeper look at the approach and it looks good. So long as the set of properties applicable under each of some parent's enum values are mutually exclusive, this approach should work well as far as I can tell.